docs/src/tut/wfs/index.md

```diff
@@ -0,0 +1,1343 @@
+	To create and retain these files, run `reach scaffold` before running `reach run`.
+
+* Line 24: The process checks the new image for vulnerabilities.
+
+    ![Checking the new image for vulnerabilities](/tut/wfs/reach-run-6.png)
+
+* Line 27: The process runs your app in a container built from the image.
+
+	![App Container](/tut/wfs/reach-run-7.png)
+
+* Line 30-31: Your application outputs these messages.
+
+<hr style="background-color:#198754;opacity:1;height:6px;"/>
+</span>
+
+# {#wfs-7} Pass an argument
+
+This section shows you how to tell your DApp whether to run as the seller or the buyer.
+You do this by passing `role` as a command-line argument (e.g. `reach run index seller`).
+You cannot do this by passing `role` as a custom environment variable (e.g. `ROLE=seller reach run`) because the [reach](https://github.com/reach-sh/reach-lang/blob/master/reach) script exports only a pre-determined list of environment variables (including `REACH_CONNECTOR_MODE`) within the Docker container where it runs your DApp.
+Follow these directions to pass `role` as a command-line argument:
+
+1. In `index.mjs`, find the following line:
+
+    ``` js nonum
+    const role = 'seller';
+    ```
+
+1. Replace it with the following, and save the file:
+
+    ``` js nonum
+    if (process.argv.length < 3 || ['seller', 'buyer'].includes(process.argv[2]) == false) {
+      console.log('Usage: reach run index [seller|buyer]');
+      process.exit(0);
+    }
+    const role = process.argv[2];
+    ```
+
+1. Open two terminals (i.e. shells):
+
+    ![Open two terminals](/tut/wfs/terminals-empty.png)
+
+1. Change directory in both terminals:
+
+    ``` nonum
+    $ ~/reach/wisdom/
+    ```
+
+1. Run the following command in both terminals:
+
+    ``` nonum
+    $ export REACH_CONNECTOR_MODE=ALGO
+    ```
+
+1. In the Seller Terminal, run your DApp as the seller:
+
+    ``` nonum
+    $ ./reach run index seller
+    ```
+
+    When you pass arguments to `./reach run`, the first one must be the name of the `index.rsh` file without the extension (i.e. `index`) as seen above. 
+    
+    Application output should resemble the following:
+
+    ``` nonum
+    The consensus network is ALGO.
+    Your role is seller.
+    ```
+
+1. In the Buyer Terminal, run your DApp as the buyer:
+
+    ``` nonum
+    $ ./reach run index buyer
+    ```
+
+    Output should resemble the following:
+
+    ``` nonum
+    The consensus network is ALGO.
+    Your role is buyer.
+    ```
+
+# {#wfs-8} Add a Node.js package
+
+This section shows you how to leverage the [yargs](https://www.npmjs.com/package/yargs) Node.js package to manage command-line arguments as an alternative to using the simple technique described above.  
+
+This section is **optional**.
+If you do **not** want to add a Node.js package to your DApp at this time, you can skip to the next section, continuing to use the following commands throughout the remainder of this module:
```

Comment:
I don't think we should have optional sections in the main flow. I think that this section could be a totally separate "guide" about "How to use more packages" and it will reference this tutorial OR it could be put after OR it could not be optional